launchdarkly-server-sdk/src/migrations/migrator.rs

```diff
@@ -0,0 +1,317 @@
+pub struct MigratorBuilder {
+    client: Client,
+    read_execution_order: ExecutionOrder,
+    measure_latency: bool,
+    measure_errors: bool,
+
+    read_config: Option<MigrationConfig>,
+    write_config: Option<MigrationConfig>,
+}
+
+impl MigratorBuilder {
+    /// Create a new migrator builder instance with the provided client.
+    pub fn new(client: Client) -> Self {
+        MigratorBuilder {
+            client,
+            read_execution_order: ExecutionOrder::Parallel,
+            measure_latency: true,
+            measure_errors: true,
+            read_config: None,
+            write_config: None,
+        }
+    }
+
+    /// The read execution order influences the parallelism and execution order for read operations
+    /// involving multiple origins.
+    pub fn read_execution_order(mut self, order: ExecutionOrder) -> Self {
+        self.read_execution_order = order;
+        self
+    }
+
+    /// Enable or disable latency tracking for migration operations. This latency information can
+    /// be sent upstream to LaunchDarkly to enhance migration visibility.
+    pub fn track_latency(mut self, measure: bool) -> Self {
+        self.measure_latency = measure;
+        self
+    }
+
+    /// Enable or disable error tracking for migration operations. This error information can be
+    /// sent upstream to LaunchDarkly to enhance migration visibility.
+    pub fn track_errors(mut self, measure: bool) -> Self {
+        self.measure_errors = measure;
+        self
+    }
+
+    /// Read can be used to configure the migration-read behavior of the resulting
+    /// [Migrator] instance.
+    ///
+    /// Users are required to provide two different read methods -- one to read from the old migration origin, and one
+    /// to read from the new origin. Additionally, customers can opt-in to consistency tracking by providing a
+    /// comparison function.
+    ///
+    /// Depending on the migration stage, one or both of these read methods may be called.
+    pub fn read(
+        mut self,
+        old: MigrationImplFn,
+        new: MigrationImplFn,
+        compare: MigrationComparisonFn,
+    ) -> Self {
+        self.read_config = Some(MigrationConfig { old, new, compare });
+        self
+    }
+
+    /// Write can be used to configure the migration-write behavior of the resulting
+    /// [Migrations::Migrator] instance.
+    ///
+    /// Users are required to provide two different write methods -- one to write to the old
+    /// migration origin, and one to write to the new origin. Not every stage requires
+    ///
+    /// Depending on the migration stage, one or both of these write methods may be called.
+    pub fn write(
+        mut self,
+        old: MigrationImplFn,
+        new: MigrationImplFn,
+        compare: MigrationComparisonFn,
+    ) -> Self {
+        self.write_config = Some(MigrationConfig { old, new, compare });
+        self
+    }
+
+    // TODO: Do we make this a real error type?
```

Comment:
(TODO) 

Is the only error this read config error so far? 

Is there something actionable they could do in code to fix it?

Reply:
These todos are for the length of the feature branch, so we don't have to block on this, but we should discuss it.

I don't think there is going to be much you can do programmatically. I mean, theoretically you could inspect the error and realize you didn't provide a read config and so you add one. But really, that's insane IMO which is why I just went with a string.

What do you think?

Reply:
Yea, I think that's fine.